### PR TITLE
switch CI to github actions

### PR DESCRIPTION
**Pull request by jmtcsngr**
Stealing most of scaffolding from https://github.com/wtsi-npg/npg_qc

### CODE REVIEW

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    - name: Run install scripts
+      run: |
+          mkdir -p "$NPM_CONFIG_PREFIX"
+          ${GITHUB_WORKSPACE}/scripts/before_install.sh
+          export PATH="$NPM_CONFIG_PREFIX/bin:$PATH"
+          ${GITHUB_WORKSPACE}/scripts/install.sh
+
+    - name: run script
+      run: |
+          export PATH="$HOME/miniconda/bin:$PATH"
```
Comment (kjsanger) [inline .github/workflows/testing.yml:57]: This PATH setting is redundant and can be removed (is done by Conda activation)

File: .github/workflows/testing.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      NPM_CONFIG_PREFIX: "$HOME/.npm-global"
+
+    name: Node ${{ matrix.node }} on ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Initialize Miniconda
+      run: |
+          echo 'source $CONDA/etc/profile.d/conda.sh' >> "$HOME/.bash_profile"
+    # Caching conda
```
Comment (kjsanger) [inline .github/workflows/testing.yml:27]: Probably not necessary to comment?

File: .github/workflows/testing.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+          conda info
+          conda create -y -n "$CONDA_TEST_ENV"
+          conda install -y -n "$CONDA_TEST_ENV" nodejs=="${{ matrix.node }}"
+      env:
+        CONDA_TEST_ENV: test-environment
+
+    - name: Run install scripts
+      run: |
+          mkdir -p "$NPM_CONFIG_PREFIX"
+          ${GITHUB_WORKSPACE}/scripts/before_install.sh
```
Comment (kjsanger) [inline .github/workflows/testing.yml:51]: Do we need a "before install" and "install"? These are terms from Travis CI's pipeline steps, which don't have equivalents in Actions. Both of these scripts install things, so simpler to just have one? Or even better perhaps, put the 3 lines from those 2 scripts directly into this step?

**Review by kjsanger — CHANGES_REQUESTED** [review]

File: .github/workflows/testing.yml (hunk 3 above)
Comment (jmtcsngr) [reply .github/workflows/testing.yml:51]: Both ways work for me. Will update things to keep it moving.

**Comment by jmtcsngr** [comment]
Thanks for the review @kjsanger, I pushed the changes.

**Pull request merged**